x/tokenomics/keeper/settle_pending_claims_context.go

```diff
@@ -218,7 +219,13 @@ func (sctx *settlementContext) cacheSupplier(
 	supplierOperatorAddress string,
 	serviceId string,
 ) error {
-	if _, ok := sctx.supplierMap[supplierOperatorAddress]; ok {
+	if idx, ok := sctx.supplierMap[supplierOperatorAddress]; ok {
+		cachedSupplier := sctx.settledSuppliers[idx]
+
+		// Supplier is cached, ensure that it has a service configuration corresponding
```

Comment:
Why does this not happen (or do we not do it) at the time the supplier is cached?

Seems more correct that way.